plugins/modules/aci_bd.py

```diff
@@ -458,22 +595,55 @@ def main():
             unkMacUcastAct=l2_unknown_unicast,
             unkMcastAct=l3_unknown_multicast,
             nameAlias=name_alias,
+            enableRogueExceptMac=enable_rogue_except_mac,
+            hostBasedRouting=host_based_routing,
+            intersiteBumTrafficAllow=allow_intersite_bum_traffic,
+            intersiteL2Stretch=allow_intersite_l2_stretch,
+            ipv6McastAllow=allow_ipv6_multicast,
+            llAddr=link_local_address,
+            mcastARPDrop=multicast_arp_drop,
+            vmac=vmac,
+            OptimizeWanBandwidth=optimize_wan_bandwidth,
         )
 
         if ipv6_l3_unknown_multicast is not None:
             class_config["v6unkMcastAct"] = ipv6_l3_unknown_multicast
 
-        aci.payload(
-            aci_class="fvBD",
-            class_config=class_config,
-            child_configs=[
-                {"fvRsCtx": {"attributes": {"tnFvCtxName": vrf}}},
-                {"fvRsIgmpsn": {"attributes": {"tnIgmpSnoopPolName": igmp_snoop_policy}}},
-                {"fvRsBDToNdP": {"attributes": {"tnNdIfPolName": ipv6_nd_policy}}},
-                {"fvRsBdToEpRet": {"attributes": {"resolveAct": endpoint_retention_action, "tnFvEpRetPolName": endpoint_retention_policy}}},
-                {"fvRsBDToProfile": {"attributes": {"tnL3extOutName": route_profile_l3out, "tnRtctrlProfileName": route_profile}}},
-            ],
-        )
+        child_configs = [
+            {"fvRsCtx": {"attributes": {"tnFvCtxName": vrf}}},
+            {"fvRsIgmpsn": {"attributes": {"tnIgmpSnoopPolName": igmp_snoop_policy}}},
+            {"fvRsMldsn": {"attributes": {"tnMldSnoopPolName": mld_snoop_policy}}},
+            {"fvRsBDToNdP": {"attributes": {"tnNdIfPolName": ipv6_nd_policy}}},
+            {"fvRsBdToEpRet": {"attributes": {"resolveAct": endpoint_retention_action, "tnFvEpRetPolName": endpoint_retention_policy}}},
+            {"fvRsBDToProfile": {"attributes": {"tnL3extOutName": route_profile_l3out, "tnRtctrlProfileName": route_profile}}},
+        ]
+
+        if igmp_policy:
+            igmp_policy_tdn = "uni/tn-{0}/igmpIfPol-{1}".format(tenant, igmp_policy)
+            child_configs.append({"igmpIfP": {"attributes": {}, "children": [{"igmpRsIfPol": {"attributes": {"tDn": igmp_policy_tdn}}}]}})
+        if vlan:
+            child_configs.append({"fvAccP": {"attributes": {"encap": vlan}}})
+        if monitoring_policy:
+            child_configs.append({"fvRsABDPolMonPol": {"attributes": {"tnMonEPGPolName": monitoring_policy}}})
+        if first_hop_security_policy:
+            child_configs.append({"fvRsBDToFhs": {"attributes": {"tnFhsBDPolName": first_hop_security_policy}}})
+        if pim_source_filter or pim_destination_filter:
+            pim_bd = {"pimBDP": {"attributes": {}, "children": []}}
+            pim_filter_pol = {"pimBDFilterPol": {"attributes": {}, "children": []}}
+            if pim_source_filter:
```

Comment:
in coverage this condition is never false, can we add a test case with only pim_source_filter defined and another one with only pim_source_filter defined?

Reply:
I have added a test case for this, however I am getting some strange behaviour when setting only pim source filter or pim destination filter and not both. APIC seems to ignore this config when not setting both filters but I am looking into possible solutions.

Reply:
Fixed and added test cases